File: 백준/Silver/1343. 폴리오미노/폴리오미노.py
import sys

check = list(sys.stdin.readline().strip())
result = ""
count = 0
flag = True
def func(result, count):
    if count == 2:
      result += "BB"
    elif count >= 4 and count % 4 == 0:
      result += "AAAA" * (count//4)
    elif count >= 4 and count % 4 == 2:
      result += "AAAA" * (count//4)
      result += "BB"
    return result

for i in range(len(check)):
  if check[i] == ".":
    if count % 2 == 1: 
        flag = False 
    if count != 0:
      result = func(result, count)
      count = 0
      result += "."
    else:
      result += "."
  else: count+=1
if count == 1: flag = False
if count % 2 == 1: 
        flag = False
if flag == False:
  print(-1)
else:
  result = func(result, count)
  print(result)

File: 백준/Silver/1343. 폴리오미노/폴리오미노.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        String str = bf.readLine();
        StringBuilder sb = new StringBuilder();
        int count = 0;
        boolean check = false;

        for (int i = 0; i < str.length(); i++) {
            char word = str.charAt(i);
            if (word == 'X') {
                count++;
                if (count == 4) {
                    sb.append("AAAA");
                    count = 0;
                }
            } else if (word == '.') {
                if (count == 4) {
                    sb.append("AAAA");
                    count = 0;
                } else if (count == 2) {
                    sb.append("BB");
                    count = 0;
                } else if (count != 0){
                    System.out.println(-1);
                    check = true;
                    break;
                }
                sb.append(".");
            }
        }

        if(count == 2){
            sb.append("BB");
            System.out.println(sb.toString());
        }else if (!check && count != 0){
            System.out.println(-1);
        }else if (!check){
            System.out.println(sb.toString());
        }


    }

}